executor/src/witgen/mod.rs

```diff
@@ -101,10 +154,30 @@ impl<'a, 'b, T: FieldElement> WitnessGenerator<'a, 'b, T> {
             self.analyzed,
             self.fixed_col_values,
             self.external_witness_values,
+            self.challenges,
         );
         let identities = self
             .analyzed
-            .identities_with_inlined_intermediate_polynomials();
+            .identities_with_inlined_intermediate_polynomials()
+            .into_iter()
+            .filter(|identity| {
+                let mut keep = true;
+                identity.pre_visit_expressions(&mut |expr| {
```

Comment:
There is the `expr_any` function, but only for Expression itself. We could lift it into the `ExpressionVisitable` trait.

Reply:
Nice, done!

Just wondering, why not have an `iter_expressions()` that returns an iterator? Seems more flexible? I think that would be for a later PR though, because it can't be implemented by a default implementation.